### PR TITLE
Ui public

### PR DESCRIPTION
**Pull request by ChrisHirsch**
Added new functionality of ui_public_target which allows binding a minion UI box to the "public" ip of the box

### CODE REVIEW

File: consul/files/consul.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
   {% if is_ui %}
-  "ui_dir": "{{ home_dir }}/dist",
+  "ui_dir": "{{ ui_install_path }}/dist",
+     {% if ui_public_target %}
+  "addresses" : {
+    "http": "{{ salt['network.interfaces']()['eth0']['inet'][0]['address'] }}"
```
Comment (lextoumbourou) [inline consul/files/consul.json:6]: No sure if there are any guarantees that eth0 will be the public interface. Thinking it should probably be configurable? What do you think?

**Comment by lextoumbourou** [comment]
Going to accept these PRs for now but make a couple of changes to them immediately. Feel free to open an issue if you are unhappy with my changes.

Thanks heaps!

File: consul/files/consul.json (hunk 1 above)
Comment (ChrisHirsch) [inline consul/files/consul.json:6]: Agreed. Do you have any good ideas on how to do that? Maybe default to eth0 and then let it be configurable via a pillar?

Comment (lextoumbourou) [inline consul/files/consul.json:6]: Yeah, this sounds good to me. I'll look at it.

**Pull request merged**